### ISSUE TITLE
CA1063 should not warn when checking if the instance is disposed.

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn-analyzers#2091]
```cs
public class C : IDisposable
{
    private bool disposed;

    public void Dispose() // CA1063 warns here
    {
        if (this.disposed)
        {
            return;
        }

        this.disposed = true;
        this.Dispose(true);
        GC.SuppressFinalize(this);
    }

    protected virtual void Dispose(bool disposing)
    {
    }
}
```

**Comment by mavasani** [comment]
This is by design, and as per the _current_ .NET Framework dispose pattern guidelines: https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implementing-the-dispose-pattern-for-a-base-class. You should move the operations on `disposed` field within the `Dispose(bool)` method.
